### PR TITLE
Adds Mecha Jackpacks

### PR DESCRIPTION
**Pull request by Belsima**
MECHA JETPACKS HERE WE COME

Sprite from /vg/ (edited).

image

Tested and functional! I'm flyin'!

### CODE REVIEW

File: code/game/mecha/equipment/tools/tools.dm [hunk 1]
```diff
@@ -57,6 +57,7 @@
 				M.adjustOxyLoss(round(dam_force/2))
 				M.updatehealth()
 				occupant_message("<span class='warning'>You squeeze [target] with [src.name]. Something cracks.</span>")
+				playsound(src.loc, "fracture", 5, 1, -2) //CRACK
```
Comment (Belsima) [inline code/game/mecha/equipment/tools/tools.dm]: just added the CRACK that you hear

File: code/game/mecha/equipment/tools/tools.dm [hunk 2]
```diff
@@ -1186,3 +1187,99 @@
...
+	name = "ion jetpack"
+	desc = "Using directed ion bursts and cunning solar wind reflection technique, this device enables controlled space flight."
+	icon_state = "mecha_jetpack"
+	equip_cooldown = 5
+	energy_drain = 50
+	var/wait = 0
+	var/datum/effect/effect/system/ion_trail_follow/ion_trail
+
+
+	can_attach(obj/mecha/M as obj)
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 3]
```diff
@@ -1186,3 +1187,99 @@
...
+		if(!(locate(src.type) in M.equipment) && !M.proc_res["dyndomove"])
+			return ..()
+
+	detach()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 4]
```diff
@@ -1186,3 +1187,99 @@
...
+		..()
+		chassis.proc_res["dyndomove"] = null
+		return
+
+	attach(obj/mecha/M as obj)
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 5]
```diff
@@ -1186,3 +1187,99 @@
...
+		..()
+		if(!ion_trail)
+			ion_trail = new
+		ion_trail.set_up(chassis)
+		return
+
+	proc/toggle()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 6]
```diff
@@ -1186,3 +1187,99 @@
...
+		if(!chassis)
+			return
+		!equip_ready? turn_off() : turn_on()
+		return equip_ready
+
+	proc/turn_on()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 7]
```diff
@@ -1186,3 +1187,99 @@
...
+		set_ready_state(0)
+		chassis.proc_res["dyndomove"] = src
+		ion_trail.start()
+		occupant_message("Activated")
+		log_message("Activated")
+
+	proc/turn_off()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 8]
```diff
@@ -1186,3 +1187,99 @@
...
+		set_ready_state(1)
+		chassis.proc_res["dyndomove"] = null
+		ion_trail.stop()
+		occupant_message("Deactivated")
+		log_message("Deactivated")
+
+	proc/dyndomove(direction)
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 9]
```diff
@@ -1186,3 +1187,99 @@
...
+			chassis.use_power(energy_drain)
+			if(!chassis.pr_inertial_movement.active())
+				chassis.pr_inertial_movement.start(list(chassis,direction))
+			else
+				chassis.pr_inertial_movement.set_process_args(list(chassis,direction))
+			do_after_cooldown()
+			return 1
+		return 0
+
+	action_checks()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 10]
```diff
@@ -1186,3 +1187,99 @@
...
+		if(equip_ready || wait)
+			return 0
+		if(energy_drain && !chassis.has_charge(energy_drain))
+			return 0
+		if(chassis.check_for_support())
+			return 0
+		return 1
+
+	get_equip_info()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 11]
```diff
@@ -1186,3 +1187,99 @@
...
+		if(!chassis) return
+		return "<span style=\"color:[equip_ready?"#0f0":"#f00"];\">*</span>&nbsp;[src.name] \[<a href=\"?src=\ref[src];toggle=1\">Toggle</a>\]"
+
+
+	Topic(href,href_list)
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

File: code/game/mecha/equipment/tools/tools.dm [hunk 12]
```diff
@@ -1186,3 +1187,99 @@
...
+		..()
+		if(href_list["toggle"])
+			toggle()
+
+	do_after_cooldown()
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: Absolute pathing.

**Comment by Belsima** [comment]
uhhh all the 'absolute pathing' is across all the mech file i dunno how to fix it all

**Comment by Atermonera** [comment]
It's changing stuff like `can_attach()` to `/obj/item/mecha_parts/mecha_equipment/tool/jetpack/can_attach()`
Part of the coding standards, so that it's absolutely explicit about what the code is attached to.

**Comment by Belsima** [comment]
Will be remaking.

**Pull request closed without merging**